ui-tests/src/test/java/org/openmrs/reference/TransferToWardServiceTest.java

```diff
@@ -11,42 +11,41 @@
 
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
-import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
+import org.openmrs.reference.page.FindPatientPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
 import org.openmrs.uitestframework.test.TestData;
 
-import static org.junit.Assert.assertTrue;
-
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
 
 public class TransferToWardServiceTest extends ReferenceApplicationTestBase {
 
     private static final String INPATIENT_WARD = "Inpatient Ward";
-
     private static final String ISOLATION_WARD = "Isolation Ward";
-
     private TestData.PatientInfo testPatient;
 
     @Before
     public void createTestData() throws Exception {
         testPatient = createTestPatient();
-        String visitType = TestData.getAVisitType();
-        String visit = new TestData.TestVisit(testPatient.uuid, visitType, getLocationUuid(homePage)).create();
     }
 
-    @Ignore("TODO fix, inpatient_ward doesn't exisit on the page, fix workflow")
     @Test
-    public void transferToWardServiceTest() throws Exception {
-
-        ClinicianFacingPatientDashboardPage patientDashboardPage = homePage.goToActiveVisitPatient();
+    @Category(BuildTests.class)
+    public void transferToWardServiceTest() {
+        FindPatientPage findPatientPage = homePage.goToFindPatientRecord();
+        findPatientPage.enterPatient(testPatient.identifier);
+        findPatientPage.waitForPageToLoad();
+        PatientVisitsDashboardPage patientVisitsDashboardPage = findPatientPage.clickOnFirstPatient().startVisit();
 
-        patientDashboardPage.goToAdmitToInpatient().confirm(INPATIENT_WARD);
-        patientDashboardPage.waitForPage();
+        patientVisitsDashboardPage.goToAdmitToInpatient().confirm(INPATIENT_WARD);
 
-        patientDashboardPage.goToTransferToWardServicePage().confirm(ISOLATION_WARD);
-        patientDashboardPage.waitForPage();
+        patientVisitsDashboardPage.goToTransferToWardServicePage().confirm(ISOLATION_WARD);
+        patientVisitsDashboardPage.waitForPage();
 
-        assertTrue(patientDashboardPage.containsText("Outpatient"));
+        assertThat(patientVisitsDashboardPage.getEncountersCount(), is(2));
```

Comment:
please add a more specific assertion to ensure the exact encounter names for the transfers are displayed

Reply:
> add a more specific assertion to ensure the exact encounter names for the transfers are displayed

I seem not to get this well!

Reply:
> add a more specific assertion to ensure the exact encounter names for the transfers are displayed

I think the suggested assertion is good enough. Why ? Because for each encounter, the encounter name is checked (affirmed) whether its displayed as at LN- [43](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/804c673b1aebc0601eba22330a7d0f612bafce98/ui-tests/src/test/java/org/openmrs/reference/TransferToWardServiceTest.java#L43) and LN-[45](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/804c673b1aebc0601eba22330a7d0f612bafce98/ui-tests/src/test/java/org/openmrs/reference/TransferToWardServiceTest.java#L45). Its from this point that we rather assert that the number of the test encounter is 2 ie `assertThat(patientVisitsDashboardPage.getEncountersCount(), is(2));`

Reply:
does the page list the exact encounter names! if yes. please add extra assertions to validate that besides the size.

Reply:
> add extra assertions to validate that besides the size. the page maybe found with more encounters and we want to validate specific ones

Interesting !! Have wrapped a response on this in the commit with id `b444bbd`